pkg/clients/iam/iamusergroupattachment.go

```diff
@@ -0,0 +1,35 @@
+package iam
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+
+	"github.com/crossplane/provider-aws/apis/identity/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// UserGroupAttachmentClient is the external client used for UserGroupAttachment Custom Resource
+type UserGroupAttachmentClient interface {
+	AddUserToGroupRequest(*iam.AddUserToGroupInput) iam.AddUserToGroupRequest
+	RemoveUserFromGroupRequest(*iam.RemoveUserFromGroupInput) iam.RemoveUserFromGroupRequest
+	ListGroupsForUserRequest(*iam.ListGroupsForUserInput) iam.ListGroupsForUserRequest
+}
+
+// NewUserGroupAttachmentClient creates new RDS RDSClient with provided AWS Configurations/Credentials
+func NewUserGroupAttachmentClient(ctx context.Context, credentials []byte, region string, auth awsclients.AuthMethod) (UserGroupAttachmentClient, error) {
+	cfg, err := auth(ctx, credentials, awsclients.DefaultSection, region)
+	if cfg == nil {
+		return nil, err
+	}
+	return iam.New(*cfg), err
+}
+
+// LateInitializeUserGroup fills the empty fields in v1alpha1.UserGroupAttachmentParameters with
+// the values seen in iam.AttachedGroup.
+func LateInitializeUserGroup(in *v1alpha1.IAMUserGroupAttachmentParameters, group *iam.Group) {
+	if group == nil {
+		return
+	}
+	in.GroupName = awsclients.LateInitializeString(in.GroupName, group.GroupName)
```

Comment:
This method late-initialises just one field. Can we do that in the controller itself?